### PR TITLE
QueryJetpackModules: Make the siteId prop be optional

### PR DESCRIPTION
**Pull request by oskosk**
* This PR updates the `QueryJetpackModules` component to declare its `siteId` prop as optional to only request data if the `siteId` prop is not `null` when mounting.

### CODE REVIEW

File: client/components/data/query-jetpack-modules/README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ export default function MyJetpackModulesList( { jetpackModules } ) {
 
 <table>
 	<tr><th>Type</th><td>Number</td></tr>
-	<tr><th>Required</th><td>Yes</td></tr>
+	<tr><th>Required</th><td>No</td></tr>
```
Comment (tyxla) [inline client/components/data/query-jetpack-modules/README.md]: Let's keep this one "Yes". Although the `siteId` can be `null` and it's not required in the context of `PropTypes`, this query component can't function without it.

File: client/components/data/query-jetpack-connection/index.jsx [hunk 2]
```diff
@@ -12,13 +12,15 @@ import { requestJetpackConnectionStatus } from 'state/jetpack/connection/actions
...
 	static propTypes = {
-		siteId: PropTypes.number.isRequired,
+		siteId: PropTypes.number,
 		requestingJetpackConnectionStatus: PropTypes.bool,
 		requestJetpackConnectionStatus: PropTypes.func
 	};
 
 	componentWillMount() {
-		this.request( this.props );
+		if ( this.props.siteId !== null ) {
```
Comment (tyxla) [inline client/components/data/query-jetpack-connection/index.jsx]: Let's do that check in `this.request` instead - see https://github.com/Automattic/wp-calypso/blob/master/client/components/data/query-jetpack-user-connection/index.jsx#L33 for example.

**Comment by Copons** [comment]
Damn I was freaking out as I thought I was having a bad dejavu here after reviewing #15365 😱 

Anyway, LGTM! 👍

**Comment by lancewillett** [comment]
Hi @Copons @oskosk ready to merge still?

**Comment by alisterscott** [comment]
Is this also ready for merge still?

**Comment by tyxla** [comment]
This one covers just a small percentage of all occurrences. After a brief discussion with @oskosk on Slack, we decided to close this one for now, and we'll revisit in the future to cover more cases.

**Pull request closed without merging**